services/streammanager/kafka/kafkamanager.go

```diff
@@ -281,7 +294,20 @@ func NewProducer(destConfigJSON interface{}, o Opts) (*producerImpl, error) { //
 	if err != nil {
 		return nil, err
 	}
-	return &producerImpl{p: p, timeout: o.Timeout}, nil
+	convertToAvro := destConfig.ConvertToAvro
+	avroSchemas := destConfig.AvroSchemas
+	var codecs map[string]goavro.Codec
+	if convertToAvro {
+		codecs = make(map[string]goavro.Codec, len(avroSchemas))
+		for _, avroSchema := range avroSchemas {
+			newCodec, err := goavro.NewCodec(avroSchema.Schema)
+			if err != nil {
+				return nil, fmt.Errorf("unable to create codec for schemaId:%+v, with error: %w", avroSchema.SchemaId, err)
+			}
+			codecs[avroSchema.SchemaId] = *newCodec
```

Comment:
What do we expect the `SchemaId` to be? Is it going to be an `int`? Or a `uuid` or something else? In any case we should at least check that it's not an empty string, right?

Reply:
It can be any of `int` or `UUID`. As it is customer-specific, I don't think it will create any issues.